### PR TITLE
Set `USER` env variable only if it is not already set.

### LINKED ISSUE #1059: [Bug] os.environ["USER"] is set unconditionally in hydra/hydra/test_utils/test_utils.py
Since os.environ["USER"] is set and hydra-core installs a pytest plugin any other pytest tests that use the $USER environment variable will get overridden to use "test_user" as the user. I think this should probably be a conditional check or something. We had a test start failing that was using $USER and it took some time to figure out why that was being set to test_user.

### PR DESCRIPTION
**Pull request by shagunsodhani**
Fixes #1059.

## Motivation

Set `USER` env variable only if it is not already set.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebookresearch/hydra/blob/master/CONTRIBUTING.md)?

Yes

## Test Plan

CI

## Related Issues and PRs

#1059

### CODE REVIEW

**Review by omry — CHANGES_REQUESTED** [review]
can you remove this and see which tests are breaking? we can probably just mock the environment variable on the specific tests.

**Comment by shagunsodhani** [comment]
Will get to it in a couple of hours.

**Comment by shagunsodhani** [comment]
ahh I think I should be explicitly unsetting the `USER` env variable if it was not originally set. I was earlier relying on the python context but then I realized that `os.environ` is a global object.

File: tests/test_examples/test_tutorials_basic.py [hunk 1]
```diff
@@ -283,12 +284,20 @@ def test_sweeping_example(
 def test_advanced_ad_hoc_composition(
     tmpdir: Path, args: List[str], expected: Any
 ) -> None:
+
+    # CircleCI does not have the environment variable USER.
+    should_remove_variable = False
```
Comment (omry) [inline tests/test_examples/test_tutorials_basic.py:289]: Use monkeypatch.
```python
def test_advanced_ad_hoc_composition(
    minkeypatch: Any, tmpdir: Path, args: List[str], expected: Any
) -> None:
  monkeypatch.setenv("USER", "test_user")
  ...
```

**Review by omry — CHANGES_REQUESTED** [review]

**Comment by omry** [comment]
Also:
1. Add a news fragment.
2. Cherry pick to the 1.0 branch.

File: tests/test_examples/test_tutorials_basic.py (hunk 1 above)
Comment (shagunsodhani) [reply tests/test_examples/test_tutorials_basic.py:289]: Ohh sweet :)

**Comment by shagunsodhani** [comment]
> Also:
> 
> 1. Add a news fragment.
> 2. Cherry pick to the 1.0 branch.

For 1, isnt this an internal change? I mean the end-user doesnt (

**Comment by shagunsodhani** [comment]
oops sorry - accidentally closed it :)

**Comment by omry** [comment]
We do, if anything we do not update news in master.
I am planning to eventually release 1.0.4 with some minor patches.

**Pull request merged**